packages/react/src/use-table/helpers.tsx

```diff
@@ -192,10 +192,14 @@ const recordToRow = (spec: Pick<TableSpec, "fieldMetadataTree" | "targetColumns"
     id: record.id,
   };
 
-  for (const targetColumn of spec.targetColumns) {
+  const customCellColumnHeaders = spec.targetColumns.filter(isCustomCellColumn).map((column) => column.header ?? "");
+  for (const [i, targetColumn] of spec.targetColumns.entries()) {
     if (isCustomCellColumn(targetColumn)) {
       const CellComponent = targetColumn.render;
-      row[targetColumn.header] = <CellComponent record={record} index={index} />;
+      // Index is included here to allow for multiple custom cells with the same header
+      const isDuplicate = customCellColumnHeaders.filter((header) => header === targetColumn.header).length > 1;
```

Comment:
This `recordToRow` function will run on each record. Is there a way to use a map or other approaches to reduce the need to do the `filter` in each for loop?

Reply:
Good call, updated to make a map of occurrences and avoid repeated filtering